File: mobile/library/swift/StreamClientImpl.swift
@_implementationOnly import EnvoyEngine
import Foundation

/// Envoy implementation of StreamClient.
final class StreamClientImpl: NSObject {
  private let engine: EnvoyEngine

  init(engine: EnvoyEngine) {
    self.engine = engine
    super.init()
  }
}

extension StreamClientImpl: StreamClient {
  func newStreamPrototype() -> StreamPrototype {
    return StreamPrototype(engine: self.engine)
  }
}

File: mobile/library/kotlin/io/envoyproxy/envoymobile/StreamClientImpl.kt
package io.envoyproxy.envoymobile

import io.envoyproxy.envoymobile.engine.EnvoyEngine

/** Envoy implementation of `StreamClient`. */
internal class StreamClientImpl(internal val engine: EnvoyEngine) : StreamClient {

  override fun newStreamPrototype() = StreamPrototype(engine)
}
